micro_dl/cli/generate_meta.py

```diff
@@ -43,26 +50,35 @@ def meta_generator(args):
     t is time
     p is position (FOV)
 
+    Other naming convention is:
+    img_channelname_t***_p***_z***.tif for parse_sms_name
+
     :param list args:    parsed args containing
         str input_dir:   path to input directory containing images
+        str name_parser: Function in aux_utils for parsing indices from file name
     """
-    meta_name = 'frames_meta.csv'
+    # Import name parser
+    parse_func = aux_utils.import_func('utils.aux_utils', args.name_parser)
 
     # Get all image names
     im_names = aux_utils.get_sorted_names(args.input)
     # Create empty dataframe
     frames_meta = aux_utils.make_dataframe(nbr_rows=len(im_names))
+    channel_names = []
     # Fill dataframe with rows from image names
     for i in range(len(im_names)):
-        frames_meta.loc[i] = aux_utils.get_ids_from_imname(
-            im_name=im_names[i],
-            order=args.order,
-        )
+        kwargs = {"im_name": im_names[i]}
+        if args.name_parser == 'parse_idx_from_name':
+            kwargs["order"] = args.order
+        elif args.name_parser == 'parse_sms_name':
```

Comment:
computational microscopy (SMS), cms may be?

Reply:
SMS is the naming convention Shalin's lab uses for all their datasets, that's the only reason I went with it...